### PR TITLE
Adds gamma distribution

### PR DESCRIPTION
**Pull request by kellijohnson-NOAA**
# What is the feature?
* Adds the gamma distribution for sampling

# How have you implemented the solution?
* Using codex 😁

# Does the PR impact any other area of the project, maybe another repo?
* No
Close #27

### CODE REVIEW

File: R/sample_gamma.R [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    cli::cli_abort(c(
+      "x" = "Length of {.var x} and {.var sd} must be the same",
+      "i" = "{.var x} has length {length(x)}",
+      "i" = "{.var sd} has length {length(sd)}"
+    ))
+  }
+
+  if (any(sd < 0)) {
+    cli::cli_abort(c(
+      "x" = "All values of {.var sd} must be positive",
```
Comment (Bai-Li-NOAA) [inline R/sample_gamma.R]: change to "All values of {.var sd} must be non-negative".

File: R/sample_gamma.R [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  if (any(x < 0)) {
+    cli::cli_abort(c(
+      "x" = "All values of {.var x} must be non-negative",
+      "i" = "{.var x} has values {x[x < 0]}"
+    ))
+  }
+
+  if (length(x) == 1 && length(sd) > 1) {
```
Comment (Bai-Li-NOAA) [inline R/sample_gamma.R:61]: If `sd` has length 1, it's not recycled to the length of `x`. If a user calls `sample_gamma(1:5, sd = 1)`, it returns a vector of identical values: 0.8361776 0.8361776 0.8361776 0.8361776 0.8361776. A possible fix from Gemini:
```
if (length(x) != length(sd)) {
    if (length(x) == 1) {
      x <- rep(x, length(sd))
    } else if (length(sd) == 1) {
      sd <- rep(sd, length(x))
    } else {
      cli::cli_abort(c(
        "x" = "Length of {.var x} and {.var sd} must be compatible",
        "i" = "{.var x} has length {length(x)}",
        "i" = "{.var sd} has length {length(sd)}"
      ))
    }
  }
```

File: tests/testthat/test-sample-.R [hunk 3]
```diff
@@ -93,6 +93,97 @@ test_that("sample_lognormal() returns correct error messages", {
...
+    c(1.0000000, 0.155141357)
+  )
+})
+
+## Edge handling ----
+test_that("sample_gamma() returns correct outputs for edge cases", {
+  #' @description Test that sample_gamma() returns a vector when a vector is
+  #' passed to `x` but a single value is passed to `sd`.
+  expect_equal(
+    object = length(sample_gamma(1:5, 1)),
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-sample-.R:139]: Add a test to check values from `sample_gamma(1:5, 1)`.

**Review by Bai-Li-NOAA — CHANGES_REQUESTED** [review]
Thanks, @kellijohnson-NOAA, for working on this issue. I have a few minor comments and requests regarding the recycling of `sd` when `x` is longer and some of the error messages.

File: R/sample_gamma.R (hunk 1 above)
Comment (kellijohnson-NOAA) [reply R/sample_gamma.R]: done 😀

File: R/sample_gamma.R (hunk 2 above)
Comment (kellijohnson-NOAA) [reply R/sample_gamma.R:61]: I don't particularly care for nested if else statements so I changed it to
```
  if (length(x) == 1 && length(sd) > 1) {
    x <- rep(x, length(sd))
  }
  if (length(sd) == 1 && length(x) > 1) {
    sd <- rep(sd, length(x))
  } 
  if (length(x) != length(sd)) {
    cli::cli_abort(c(
      "x" = "Length of {.var x} and {.var sd} must be compatible",
      "i" = "{.var x} has length {length(x)}",
      "i" = "{.var sd} has length {length(sd)}"
    ))
  }
```

File: tests/testthat/test-sample-.R (hunk 3 above)
Comment (kellijohnson-NOAA) [reply tests/testthat/test-sample-.R:139]: done
